extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/MicrometerSendingInterceptor.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import io.micrometer.core.instrument.Counter;
+import io.vertx.core.buffer.Buffer;
+
+final class MicrometerSendingInterceptor implements SendingInterceptor {
+
+    private final Counter onMessageSentCounter;
+    private final Counter onMessageSentBytesCounter;
+
+    MicrometerSendingInterceptor(Counter onMessageSentCounter, Counter onMessageSentBytesCounter) {
+        this.onMessageSentCounter = onMessageSentCounter;
+        this.onMessageSentBytesCounter = onMessageSentBytesCounter;
+    }
+
+    @Override
+    public void onSend(String text) {
+        onMessageSentCounter.increment();
+        onMessageSentBytesCounter.increment(text.getBytes().length);
```

Comment:
`text.getBytes()` is using the platform's default charset. I wonder if that could be a problem?

Reply:
tomorrow I'll look how Vert.x (or netty judging by metrics names here https://vertx.io/docs/vertx-micrometer-metrics/java/)  are doing this, unless they perform everything on the `Buffer`; will get back to you

Reply:
@mkouba if I tracked this code right, where this `String` comes from in Vert.x is here https://github.com/eclipse-vertx/vert.x/blob/master/vertx-core/src/main/java/io/vertx/core/http/impl/WebSocketImplBase.java#L492 which means that it is UTF-8 according to this Javadoc https://github.com/eclipse-vertx/vert.x/blob/master/vertx-core/src/main/java/io/vertx/core/buffer/Buffer.java#L96

My conclusion is that if I replace this code with `text.getBytes(StandardCharsets.UTF_8).length` is will be correct charset. Does it sound acceptable?

Reply:
done